xml/System.Diagnostics/Process.xml

```diff
@@ -862,7 +862,7 @@ The following code example creates a process that prints a file. It sets the <xr
  The event only occurs during asynchronous read operations on <xref:System.Diagnostics.Process.StandardError%2A>. To start asynchronous read operations, you must redirect the <xref:System.Diagnostics.Process.StandardError%2A> stream of a <xref:System.Diagnostics.Process>, add your event handler to the <xref:System.Diagnostics.Process.ErrorDataReceived> event, and call <xref:System.Diagnostics.Process.BeginErrorReadLine%2A>. Thereafter, the <xref:System.Diagnostics.Process.ErrorDataReceived> event signals each time the process writes a line to the redirected <xref:System.Diagnostics.Process.StandardError%2A> stream, until the process exits or calls <xref:System.Diagnostics.Process.CancelErrorRead%2A>.  
   
 > [!NOTE]
->  The application that is processing the asynchronous output should call the <xref:System.Diagnostics.Process.WaitForExit%2A> method to ensure that the output buffer has been flushed.  
+>  The application that is processing the asynchronous output should call the <xref:System.Diagnostics.Process.WaitForExit%2A> method to ensure that the output buffer has been flushed. Please note that the <xref:System.Diagnostics.Process.WaitForExit(Int32)> overload does *not* ensure the output buffer has been flushed.
```

Comment:
Looking at the source code in both [.Net Framework](http://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Process.cs,2458) and [.Net Core](https://source.dot.net/#System.Diagnostics.Process/System/Diagnostics/Process.Windows.cs,160), this is slightly inaccurate: `WaitForExit(-1)` *does* wait for the output buffer. So maybe change it to something like:

```
Please note that specifying a timeout by using the <xref:System.Diagnostics.Process.WaitForExit(Int32)> overload does *not* ensure the output buffer has been flushed.
```

Also, I think the first reference should be changed from `WaitForExit*` to `WaitForExit()`, so that it points to the right overload, not the whole method group. (And maybe that would actually be enough and the added sentence isn't necessary?)

Reply:
I agree with @svick's suggested change ("...specifying a timeout by using...").

Although the added sentence isn't necessary, it would be good to leave it for greater clarity. But please change "Please note" to "Note".

Finally, the reference should definitely be changed from WaitForExit%2A to simply WaitForExit. At build time, that resolves to the parameterless method, whereas the form with the %2A resolves to the overload page as a whole.